freva-rest/src/databrowser_api/endpoints.py

```diff
@@ -245,72 +262,57 @@ async def load_data(
     )
 
 
-@app.put(
-    "/api/databrowser/userdata/{username}",
+@app.post(
+    "/api/databrowser/userdata",
     status_code=status.HTTP_202_ACCEPTED,
     tags=["User Data"],
 )
-async def put_user_data(
-    username: str,
-    paths: Annotated[
-        Sequence[str], Query(description="Paths to the user's data to be added")
-    ],
-    facets: Annotated[
-        Mapping[str, str], Body(..., description="Facet key-value pairs for metadata")
-    ],
-    background_tasks: BackgroundTasks,
+async def post_user_data(
+    request: Annotated[AddUserDataRequestBody, Body(...)],
     current_user: TokenPayload = Depends(auth.required),
 ) -> Dict[str, str]:
-    """Add user data to the Apache Solr search system and MongoDB."""
-    if username != current_user.preferred_username:  # type: ignore
-        raise HTTPException(
-            status_code=status.HTTP_403_FORBIDDEN, detail="Permission denied"
-        )
+    """Add user metadata into the Apache Solr search system and MongoDB."""
 
     solr_instance = Solr(server_config)
-
     try:
-        validated_paths = await solr_instance._validating_userdata_paths(*paths)
-        background_tasks.add_task(
-            solr_instance.add_userdata, (username, validated_paths, facets)
+        validated_user_metadata = await solr_instance._validate_user_metadata(
+            request.user_metadata
         )
-    except FileNotFoundError as e:
-        raise HTTPException(
-            status_code=status.HTTP_400_BAD_REQUEST,
-            detail=f"Invalid paths provided: {str(e)}",
+        await solr_instance.add_user_metadata(
+            current_user.preferred_username,  # type: ignore
+            validated_user_metadata,
+            facets=request.facets
         )
-    except Exception as e:  # pragma: no cover
+    except Exception as e:
```

Comment:
I think you want to log an exception here - in case someone comes and complains that the user data won't ingest so an admin could have a look